virttest/utils_params.py

```diff
@@ -104,3 +105,91 @@ def copy_from_keys(self, keys):
+                             does not exist (defaults to False/"no").
+        :return whether option is set to 'yes'
+        :rtype: bool
+        """
+        value = self.get(key, "yes" if default else "no")
+        if value in ("yes", "on", "true"):
+            return True
+        if value in ("no", "off", "false"):
+            return False
+        raise ValueError("Cannot get boolean parameter value for %s: %s", key, value)
+
+    def get_numeric(self, key, default=0, target_type=int):
+        """
+        Get numeric value converting to integer if necessary.
+
+        :param str key: config option key
+        :param int default: default numeric value
+        :param type target_type: numeric type to return like int or float
+        :return numerical type `target_type` converted parameter value
+        :rtype: int or float
+        """
+        return target_type(self.get(key, default))
+
+    def get_list(self, key, default="", delimiter=' ', target_type=str):
+        """
+        Get a parameter value that is a character delimited list.
+
+        :param str key: parameter whose value is list
+        :param str default: default list value
+        :param str delimiter: character to split list items
+        :param type target_type: type of each item, default is string
+        :returns: empty list if if the key is not in the parameters
+        :rtype: [str]
+
+        .. note:: This an extension to the :py:func:`Params.objects` method as
+          it allows for delimiters other than white space.
+        .. seealso:: :py:func:`param_dict`
+        """
+        param_string = self.get(key, default)
+        if not param_string:
+            return []
+        else:
+            return [target_type(entry) for entry in param_string.split(delimiter)]
+
+    def get_dict(self, key, default="", delimiter=' ', need_order=False):
+        """
+        Get a param value that has the form 'name1=value1 name2=value2 ...'.
+
+        :param str key: parameter whose value is dict
+        :param str default: default dict value
+        :param str delimiter: character to split list items
+        :param bool need_order: whether to return an OrderedDict instead of
+                                a regular dict
+        :returns: empty dict if if the key is not in the parameters, a dict
+                  or an ordered dictionary if the item order is important
+        :rtype: {str: str}
+
+        This uses :py:meth:`get_list` to convert the list entries to dict.
+        """
+        if need_order:
+            result = OrderedDict()
+        else:
+            result = dict()
+        for entry in self.get_list(key, default, delimiter):
+            index = entry.find('=')
+            if index == -1:
+                raise ValueError('failed to find "=" in "{0}" (value for {1})'
+                                 .format(entry, key))
+            result[entry[:index]] = entry[index+1:]
```

Comment:
This might be tricky with human-entered values as extra spaces might be present. Do you think `strip` would be useful here? (I don't think it's needed, just asking whether you considered this)

Reply:
Not really considered since this method follows a similar approach like the list method and is stricter when it comes to config formatting. However, we can also make it more flexible to use the same approach for both methods again.

-> Will be added in another smaller PR.